bftengine/src/bftengine/messages/MessageBase.cpp

```diff
@@ -65,6 +65,23 @@ void MessageBase::shrinkToFit() {
   storageSize_ = msgSize_;
 }
 
+bool MessageBase::reallocSize(uint32_t size) {
```

Comment:
Is this being used in the current PR? Maybe you can add some comment to clarify the purpose.

Reply:
Yes, it is used. When we initiate View Change, we allocate a new ViewChangeMsg [ReplicaImp.cpp#L2260](https://github.com/vmware/concord-bft/blob/31ae2ee35413acf879ba10870137fe7dadf27bab/bftengine/src/bftengine/ReplicaImp.cpp#L2260) . If the next expected Primary does not take action during the View Change process, we need to initiate additional View Change to move to the next Primary. In this process we do not release the VeiwChangeMsg, but instead only modify it to reflect the New View [ReplicaImp.cpp#L2223](https://github.com/vmware/concord-bft/blob/31ae2ee35413acf879ba10870137fe7dadf27bab/bftengine/src/bftengine/ReplicaImp.cpp#L2223) but the information for the Sequence Numbers remains the same. We need to put the new complaints from peer Replicas, though, because they will be different from the initial ones. In order to do so we need to reallocate back to the max message size, clear the old complaints, add the new ones, and finally shrink the unused bytes and sign the new contents (without the complaints, only the View number combined with the SeqNo info).